### PR TITLE
Update Golang Jenkinsfile

### PR DESCRIPTION
**Pull request by jarifibrahim**
This PR
 - Uses new pipeline library for Golang Jenkinsfiles
 - Updates `releaseAndStage` and `releaseStageApproveAndPromote` pipeline's Jenkinsfile for Golang
 - Adds new `release` pipeline Jenkinsfile for Golang

NOTE: 
 - I've set `config runtime: 'golang'` in all the Jenkinsfile even though we do not have Golang slave images right now.  I couldn't find the correct slave image. If someone can point me to the correct Golang slave image, I'd be happy to fix that as well in this PR.
The build works with/without the `config runtime: 'golang'` declaration

- I've tested the Jenkinsfile with Golang boosters in https://github.com/golang-starters

See also https://openshift.io/openshiftio/Openshift_io/plan/detail/1265

### CODE REVIEW

**Review by piyush-garg — COMMENTED** [review]
Changes look good to me.

Few concerns
1. I think `go` is better than `golang`
2. Can you update the readme and metadata.yml for `ReleaseStage` and `ReleaseStageApproveAndPromote`pipeline also to make it consistent for all.

**Comment by jarifibrahim** [comment]
>I think go is better than golang

Aren't both the same? :wink: 

>Can you update the readme and metadata.yml for ReleaseStage and ReleaseStageApproveAndPromotepipeline also to make it consistent for all.

Done. See https://github.com/fabric8io/fabric8-jenkinsfile-library/pull/45/commits/d6cc943eeaf51714cfa45b7729f850831f1dc806

**Comment by piyush-garg** [comment]
> > I think go is better than golang
> 
> Aren't both the same? 
> 
> > Can you update the readme and metadata.yml for ReleaseStage and ReleaseStageApproveAndPromotepipeline also to make it consistent for all.
> 
> Done. See [d6cc943](https://github.com/fabric8io/fabric8-jenkinsfile-library/commit/d6cc943eeaf51714cfa45b7729f850831f1dc806)



> > I think go is better than golang
> 
> Aren't both the same? 
> 

I was saying for runtime value - `go` looks better than `golang`
> > Can you update the readme and metadata.yml for ReleaseStage and ReleaseStageApproveAndPromotepipeline also to make it consistent for all.
> 
> Done. See [d6cc943](https://github.com/fabric8io/fabric8-jenkinsfile-library/commit/d6cc943eeaf51714cfa45b7729f850831f1dc806)

**Comment by jarifibrahim** [comment]
>I was saying for runtime value - go looks better than golang

Done. See https://github.com/fabric8io/fabric8-jenkinsfile-library/pull/45/commits/de420a37786eb1ccdffe5d95cbbcd73aa235be65

**Review by piyush-garg — APPROVED** [review]

File: golang/Release/metadata.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+suggested: false
```
Comment (hrishin) [inline golang/Release/metadata.yml:1]: whats the purpose of these flags?

File: golang/Release/metadata.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+tech-preview: false
```
Comment (hrishin) [inline golang/Release/metadata.yml:2]: I guess it should be tech-preview first?

**Review by hrishin — COMMENTED** [review]
LGTM.

Thank you @jarifibrahim for the patch.

File: golang/Release/metadata.yml (hunk 1 above)
Comment (jarifibrahim) [reply golang/Release/metadata.yml:1]: This is used to show the suggested box around the pipeline in the UI

File: golang/Release/metadata.yml (hunk 2 above)
Comment (jarifibrahim) [reply golang/Release/metadata.yml:2]: Not really. The tech-preview flag is used along with the suggested flag to show the blue box arount the pipeline in the UI.

See https://github.com/fabric8-launcher/ngx-launcher/blob/36c134f3e4e0ed30a76dcb954f7a3e94af35926f/projects/ngx-launcher/src/lib/components/release-strategy-step/release-strategy-step.component.html#L26-L27

Comment (hrishin) [reply golang/Release/metadata.yml:2]: thanks!

**Review by hrishin — APPROVED** [review]

**Comment by sbose78** [comment]
Wish there were CI tests for these.

**Review by sbose78 — APPROVED** [review]

**Pull request merged**